### PR TITLE
YARN-11103. SLS cleanup after previously merged SLS refactor jiras

### PR DESCRIPTION
**Pull request by szilard-nemeth**
### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  5s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  1s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  37m 43s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 28s |  |  trunk passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  compile  |   0m 25s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 22s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 30s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 28s |  |  trunk passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javadoc  |   0m 25s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| -1 :x: |  spotbugs  |   0m 50s | [/branch-spotbugs-hadoop-tools_hadoop-sls-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/1/artifact/out/branch-spotbugs-hadoop-tools_hadoop-sls-warnings.html) |  hadoop-tools/hadoop-sls in trunk has 2 extant spotbugs warnings.  |
| +1 :green_heart: |  shadedclient  |  25m  8s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 21s |  |  the patch passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javac  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 19s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   0m 19s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/1/artifact/out/blanks-eol.txt) |  The patch has 1 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -0 :warning: |  checkstyle  |   0m 13s | [/results-checkstyle-hadoop-tools_hadoop-sls.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/1/artifact/out/results-checkstyle-hadoop-tools_hadoop-sls.txt) |  hadoop-tools/hadoop-sls: The patch generated 14 new + 23 unchanged - 24 fixed = 37 total (was 47)  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 19s |  |  hadoop-tools_hadoop-sls-jdkUbuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 generated 0 new + 19 unchanged - 6 fixed = 19 total (was 25)  |
| +1 :green_heart: |  javadoc  |   0m 17s |  |  hadoop-tools_hadoop-sls-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 generated 0 new + 19 unchanged - 6 fixed = 19 total (was 25)  |
| -1 :x: |  spotbugs  |   0m 53s | [/new-spotbugs-hadoop-tools_hadoop-sls.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/1/artifact/out/new-spotbugs-hadoop-tools_hadoop-sls.html) |  hadoop-tools/hadoop-sls generated 1 new + 1 unchanged - 1 fixed = 2 total (was 2)  |
| +1 :green_heart: |  shadedclient  |  23m 47s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  12m  4s |  |  hadoop-sls in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   | 108m 14s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-tools/hadoop-sls |
|  |  Write to static field org.apache.hadoop.yarn.sls.AMRunner.remainingApps from instance method org.apache.hadoop.yarn.sls.AMRunner.startAM()  At AMRunner.java:from instance method org.apache.hadoop.yarn.sls.AMRunner.startAM()  At AMRunner.java:[line 117] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4119 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 8ad2c5275e18 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 981827bbd38e481e26d37c4aebba051f8f7ea377 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/1/testReport/ |
| Max. process+thread count | 522 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-sls U: hadoop-tools/hadoop-sls |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m 21s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 3 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  37m 46s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 30s |  |  trunk passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  compile  |   0m 24s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 21s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 29s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 29s |  |  trunk passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javadoc  |   0m 23s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| -1 :x: |  spotbugs  |   0m 49s | [/branch-spotbugs-hadoop-tools_hadoop-sls-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/2/artifact/out/branch-spotbugs-hadoop-tools_hadoop-sls-warnings.html) |  hadoop-tools/hadoop-sls in trunk has 2 extant spotbugs warnings.  |
| +1 :green_heart: |  shadedclient  |  23m 47s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 24s |  |  the patch passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javac  |   0m 24s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 20s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   0m 20s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/2/artifact/out/blanks-eol.txt) |  The patch has 1 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -0 :warning: |  checkstyle  |   0m 15s | [/results-checkstyle-hadoop-tools_hadoop-sls.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/2/artifact/out/results-checkstyle-hadoop-tools_hadoop-sls.txt) |  hadoop-tools/hadoop-sls: The patch generated 13 new + 53 unchanged - 29 fixed = 66 total (was 82)  |
| +1 :green_heart: |  mvnsite  |   0m 22s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 21s |  |  hadoop-tools_hadoop-sls-jdkUbuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 generated 0 new + 19 unchanged - 6 fixed = 19 total (was 25)  |
| +1 :green_heart: |  javadoc  |   0m 17s |  |  hadoop-tools_hadoop-sls-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 generated 0 new + 19 unchanged - 6 fixed = 19 total (was 25)  |
| -1 :x: |  spotbugs  |   0m 56s | [/new-spotbugs-hadoop-tools_hadoop-sls.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/2/artifact/out/new-spotbugs-hadoop-tools_hadoop-sls.html) |  hadoop-tools/hadoop-sls generated 1 new + 1 unchanged - 1 fixed = 2 total (was 2)  |
| +1 :green_heart: |  shadedclient  |  23m 30s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  12m  3s |  |  hadoop-sls in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 29s |  |  The patch does not generate ASF License warnings.  |
|  |   | 106m 59s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-tools/hadoop-sls |
|  |  Write to static field org.apache.hadoop.yarn.sls.AMRunner.remainingApps from instance method org.apache.hadoop.yarn.sls.AMRunner.startAM()  At AMRunner.java:from instance method org.apache.hadoop.yarn.sls.AMRunner.startAM()  At AMRunner.java:[line 116] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4119 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 81a7671907b1 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / f4382e448deee7e74269d510e850a492cf97498a |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/2/testReport/ |
| Max. process+thread count | 593 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-sls U: hadoop-tools/hadoop-sls |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java
Comment (brumi1024) [inline hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java:56]: Does this field need to be static? Spotbugs warns because this way it isn't threadsafe if multiple AMRunner instances write this. Referencing it via AMRunner.remaningApps should make the spotbugs warning disappear but it won't make this threadsafe.

**Review by brumi1024 — COMMENTED** [review]
Thanks @szilard-nemeth for the patch. I had one minor comment.

File: hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java
Comment (szilard-nemeth) [reply hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java:56]: As the spotbugs warning is strictly related to this remainingApps field, I'd prefer to fix it with https://issues.apache.org/jira/browse/YARN-11102

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  4s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 3 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  38m  2s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 35s |  |  trunk passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  compile  |   0m 39s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 46s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  trunk passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| -1 :x: |  spotbugs  |   1m  8s | [/branch-spotbugs-hadoop-tools_hadoop-sls-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/3/artifact/out/branch-spotbugs-hadoop-tools_hadoop-sls-warnings.html) |  hadoop-tools/hadoop-sls in trunk has 3 extant spotbugs warnings.  |
| +1 :green_heart: |  shadedclient  |  25m 13s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 22s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 22s |  |  the patch passed with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javac  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 18s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   0m 18s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/3/artifact/out/blanks-eol.txt) |  The patch has 1 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -0 :warning: |  checkstyle  |   0m 13s | [/results-checkstyle-hadoop-tools_hadoop-sls.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/3/artifact/out/results-checkstyle-hadoop-tools_hadoop-sls.txt) |  hadoop-tools/hadoop-sls: The patch generated 13 new + 59 unchanged - 24 fixed = 72 total (was 83)  |
| +1 :green_heart: |  mvnsite  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 19s |  |  hadoop-tools_hadoop-sls-jdkUbuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 with JDK Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 generated 0 new + 20 unchanged - 6 fixed = 20 total (was 26)  |
| +1 :green_heart: |  javadoc  |   0m 17s |  |  hadoop-tools_hadoop-sls-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 generated 0 new + 20 unchanged - 6 fixed = 20 total (was 26)  |
| -1 :x: |  spotbugs  |   0m 54s | [/new-spotbugs-hadoop-tools_hadoop-sls.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/3/artifact/out/new-spotbugs-hadoop-tools_hadoop-sls.html) |  hadoop-tools/hadoop-sls generated 1 new + 2 unchanged - 1 fixed = 3 total (was 3)  |
| +1 :green_heart: |  shadedclient  |  25m 25s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  12m 16s |  |  hadoop-sls in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 111m 58s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-tools/hadoop-sls |
|  |  Write to static field org.apache.hadoop.yarn.sls.AMRunner.remainingApps from instance method org.apache.hadoop.yarn.sls.AMRunner.startAM()  At AMRunner.java:from instance method org.apache.hadoop.yarn.sls.AMRunner.startAM()  At AMRunner.java:[line 115] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4119 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 0999cbd810d1 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 66493fa55a3961f057106ef92cac3abc3e6c4a9c |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.14+9-Ubuntu-0ubuntu2.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/3/testReport/ |
| Max. process+thread count | 523 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-sls U: hadoop-tools/hadoop-sls |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4119/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java
Comment (9uapaw) [reply hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java:56]: AFAIU SpotBugs does not warn because it is not thread safe. A non-static int is not thread safe either, unless you make it an AtomicInteger or wrap it around a mutex (though static AND thread unsafe variables are more dangerous I agree). SpotBugs simply warns because you are incrementing a static int outside of the class that owns it, hence it warns of a possible bug.
In order to make it safer while removing the SpotBugs error:
- Make it an AtomicInteger
- Increment it via a static increment method (this is not any safer, but it would most probably remove the error message)

**Review by 9uapaw — APPROVED** [review]
Thanks for the PR @szilard-nemeth. It is straightforward and a good clean-up change. Let us wait for YARN-11102 to be merged in order not to increase spotbugs warning. I will commit it afterwards.

**Comment by szilard-nemeth** [comment]
@9uapaw 
Thanks for your comment. This is an independent change and the spotbugs warning was in place before this patch, so I suggest to commit this first then I will fix the spotbugs in the other one.
Thanks

File: hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java
Comment (brumi1024) [reply hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java:56]: However if only one class is modifying this static variable a time (or if the owner class is a singleton) there is no problem with having a static field being modified from an instance method (a basic example would be to count the number of instances of a class in a completely single threaded environment).

Comment (9uapaw) [reply hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java:56]: Indeed, this idiom has its places, the SpotBugs only warns us because it is generally not a good idea. All in all, in this case, the static integer is simply a poor architectural decision, therefore I advised in YARN-11102 to make it an instance member, thus eliminating all kinds of errors.

**Comment by brumi1024** [comment]
Thanks @szilard-nemeth for the PR, since a different PR exists for the spotbugs issue I committed this one to trunk. Fixed the indentation checkstyle issues before merging.

File: hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java
Comment (brumi1024) [reply hadoop-tools/hadoop-sls/src/main/java/org/apache/hadoop/yarn/sls/AMRunner.java:56]: Definitely agree.

**Pull request closed without merging**
